src/shell-app.c

```diff
@@ -937,7 +939,10 @@ shell_app_sync_running_state (ShellApp *app)
 
   if (app->state != SHELL_APP_STATE_STARTING)
     {
-      if (app->running_state->interesting_windows == 0)
+      /* We check if the startup completed as a workaround for apps that show
+       * a window and quickly remove it before the main window shows up, which
+       * is the case with Maps. */
+      if (app->running_state->interesting_windows == 0 && app->running_state->startup_complete)
```

Comment:
My suggestion to do it the other way is because it avoids relying on the fact that a speedwagon window will be there to transition to the STOPPED state. With your patch as it is, we'll only transition to STOPPED when a speedwagon window has been seen, which is a bit suboptimal.